File: app/models/transcription.py
"""Pydantic models for transcription data."""

from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, Field


class TranscriptionStatus(str, Enum):
    """Transcription processing status."""

    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class TranscriptionSegment(BaseModel):
    """A single segment of transcribed text with timestamps."""

    id: str
    start_time: float = Field(ge=0, description="Start time in seconds")
    end_time: float = Field(ge=0, description="End time in seconds")
    text: str


class Transcription(BaseModel):
    """Complete transcription data."""

    id: str
    status: TranscriptionStatus
    file_name: str
    file_type: str
    duration: Optional[float] = None
    language: Optional[str] = "en"
    segments: List[TranscriptionSegment] = []


class TranscriptionItem(BaseModel):
    """Simplified transcription data for list views."""

    id: str
    status: TranscriptionStatus
    file_name: str
    duration: Optional[float] = None


Transcriptions = List[TranscriptionItem]

File: ui/src/types/transcription.ts
export type Status = "pending" | "processing" | "completed" | "failed";

export interface TranscriptionItem {
  id: string;
  status: Status;
  file_name: string;
  duration?: number;
}

export interface TranscriptionSegment {
  id: string;
  start_time: number;
  end_time: number;
  text: string;
  speaker?: string;
}

export interface Transcription {
  id: string;
  status: Status;
  fileName: string;
  fileType: string;
  duration?: number;
  language?: string;
  segments: TranscriptionSegment[];
}

export interface TranscriptionUpdate {
  segments: TranscriptionSegment[];
}
